pkg/commands/start_standalone.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/pkg/dns"
+	log "github.com/sirupsen/logrus"
+)
+
+func StartStandAloneServer(address string,
+	udpLocalPort,
+	tcpLocalPort int,
+	ipv6 bool,
+	hosts map[string]string,
+	upstreamServers []string) error {
```

Comment:
That's a lot of arguments — consider using an options object instead. In fact, all the arguments are in `dns.ServerOptions`…